gui/CMakeLists.txt

```diff
@@ -61,6 +58,10 @@ CheckOptions:
         target_compile_definitions(cppcheck-gui PRIVATE CPPCHECKLIB_IMPORT TINYXML2_IMPORT)
         target_link_libraries(cppcheck-gui cppcheck-core)
     endif()
+    if(MSVC)
```

Comment:
Good catch. But I am curious why I (and several others for sure) never ran into any issues with the GUI although it was never actually disabled.

It might be related to this being used in `config.h` which is not included in every source file (and I guess it only needs to always be the first include which we also don't enforce). So that functionality was never fully working and I wonder if we should just drop the whole `_CRTDBG_MAP_ALLOC` stuff.

Reply:
Ah, I see you are generating a completely new solution which contains the GUI files as well. Nobody ever tested that and until we drop the provided solution file (which we most likely never will) I also consider this as not officially supported.

And adding this to the provided solution adds too much complexity which I am not willing to maintain - and I would also very much reject an external submission adding this as it would increase the build matrix complexity and if it needs to be adjusted it would break the build and the maintainer would end up somebody "within" project.

Reply:
Sorry, I misread. The compilation failure was known and was already addressed so I am even more baffled how it passed the CI.